src/command.c

```diff
@@ -83,33 +83,35 @@ static bool cmd_shutdown_bmda(target_s *t, int argc, const char **argv);
 const command_s cmd_list[] = {
 	{"version", cmd_version, "Display firmware version info"},
 	{"help", cmd_help, "Display help for monitor commands"},
-	{"jtag_scan", cmd_jtag_scan, "Scan JTAG chain for devices"},
-	{"swdp_scan", cmd_swdp_scan, "Scan SW-DP for devices"},
+	{"jtag_scan", cmd_jtag_scan, "Scan JTAG chain for devices: [IR_LENGTH ...]"},
+	{"swdp_scan", cmd_swdp_scan, "Scan SW-DP for devices: [TARGET_ID]"},
 	{"auto_scan", cmd_auto_scan, "Automatically scan all chain types for devices"},
-	{"frequency", cmd_frequency, "set minimum high and low times"},
+	{"frequency", cmd_frequency, "set minimum high and low times: [FREQUENCY]"},
 	{"targets", cmd_targets, "Display list of available targets"},
 	{"morse", cmd_morse, "Display morse error message"},
-	{"halt_timeout", cmd_halt_timeout, "Timeout (ms) to wait until Cortex-M is halted: (Default 2000)"},
-	{"connect_rst", cmd_connect_reset, "Configure connect under reset: (enable|disable)"},
-	{"reset", cmd_reset, "Pulse the nRST line - disconnects target: (pulse_len_ms, default 0)"},
+	{"halt_timeout", cmd_halt_timeout, "Timeout to wait until Cortex-M is halted: [TIMEOUT_MS, default 2000]"},
```

Comment:
it's only one character, but perhaps `[TIMEOUT, default 2000ms]`? that still expresses the units used (similar suggestion for the pulse length one two lines down)

Reply:
it should reduce the code size by a byte per (for 2 bytes total), given we're suggesting `[TIMEOUT_MS, default 2000]` becomes `[TIMEOUT, default 2000ms]`. Interesting point regarding it sounding like you can specify units - that's kind of a problem either way in our opinion and something that the limited number of characters to describe the command causes.

One way we could deal with this is apply our suggestion and then also add some documentation to the website that explains that the commands have fixed units so if people have a query, we can point them to the documentation that can then greatly expand on these strings - how would you feel about that?

Reply:
Fine with me. Follow-up MR: blackmagic-debug/black-magic-org#21. Wasn't sure where to place it so I just added a sentence after the example `monitor help` output.